lib/balance-tx/lib/internal/Internal/Cardano/Write/Tx.hs

```diff
@@ -665,29 +670,29 @@ serializeTx
     :: forall era. IsRecentEra era
     => Core.Tx era
     -> ByteString
-serializeTx tx = CardanoApi.serialiseToCBOR $ toCardanoApiTx @era tx
+serializeTx tx = CardanoApi.serialiseToCBOR $ toCardanoApiTx recentEra tx
 
 --------------------------------------------------------------------------------
 -- Compatibility
 --------------------------------------------------------------------------------
 
 fromCardanoApiTx
-    :: forall era. IsRecentEra era
-    => CardanoApi.Tx (CardanoApiEra era)
+    :: RecentEra era
+    -> CardanoApi.Tx (CardanoApiEra era)
     -> Core.Tx era
-fromCardanoApiTx = \case
+fromCardanoApiTx era = \case
     CardanoApi.ShelleyTx _era tx ->
-        tx
+        withRecentEra era tx
     CardanoApi.ByronTx {} ->
-        case (recentEra @era) of
+        case era of
             {}
 
 toCardanoApiTx
-    :: forall era. IsRecentEra era
-    => Core.Tx era
+    :: RecentEra era
+    -> Core.Tx era
```

Comment:
I have finalized the text of the "Singleton type RecentEra :: AP" Advice Process.

The decision would be that `forall era. IsRecentEra era` is the way to go.

Jonathan, could you take another look at the text and give additional Advice, if any?